include/seqan3/core/debug_stream.hpp

```diff
@@ -0,0 +1,202 @@
+// seqan3::debug_stream
+// ------------------------------------------------------------------
+
+/*!\brief A "pretty printer" for most SeqAn data structures and related types.
+ * \ingroup core
+ * \details
+ *
+ * You can stream to seqan3::debug_stream as you would to std::cout or std::cerr, but the debug stream has special
+ * overloads that make certain types streamable (that are not streamable to std::cout). Additionally some
+ * data structures are visualised more elaborately via the debug stream.
+ *
+ * ## Example
+ *
+ * \snippet test/snippet/core/debug_stream.cpp usage
+ *
+ * The actual underlying stream that is printed to defaults to std::cerr, but can be changed via
+ * seqan3::debug_stream_t::set_stream. You can set any kind of output stream, e.g. a std::ostringstream or a
+ * std::ofstream if you want to write to a file, but please be aware that the debug_stream never takes ownership of the
+ * underlying stream so you need to take special care that its object lifetime does not end before the debug_stream's.
+ *
+ * \snippet test/snippet/core/debug_stream.cpp set_stream
+ *
+ * In the case where you wish to print to some stream object locally, instead create you own debug stream:
+ *
+ * \snippet test/snippet/core/debug_stream.cpp set_stream2
+ */
+inline debug_stream_t debug_stream{std::cerr};
+
+// ------------------------------------------------------------------
+// overloads
+// ------------------------------------------------------------------
+
+/*!\brief All alphabets can be printed to the seqan3::debug_stream by their char representation.
+ * \tparam alphabet_t Type of the alphabet to be printed; must model seqan3::alphabet_concept.
+ * \param s The seqan3::debug_stream.
+ * \param l The alphabet letter.
+ * \relates seqan3::debug_stream_t
+ */
+template <alphabet_concept alphabet_t>
+inline debug_stream_t & operator<<(debug_stream_t & s, alphabet_t const l)
+//!\cond
+    requires !(requires (alphabet_t l) { { std::cerr << l }; })
+//!\endcond
+{
+    return s << to_char(l);
+}
+
+/*!\brief All input ranges can be printed to the seqan3::debug_stream element-wise (if their elements are printable).
+ * \tparam rng_t Type of the range to be printed; must model std::ranges::InputRange.
+ * \param s The seqan3::debug_stream.
+ * \param r The input range.
+ * \relates seqan3::debug_stream_t
+ *
+ * \details
+ *
+ * If the element type models seqan3::alphabet_concept (and is not an unsigned integer), the range is printed
+ * just as if it were a string, i.e. `std::vector<dna4>{dna4::C, dna4:G, dna4::A}` is printed as "CGA".
+ *
+ * In all other cases the elements are comma separated and the range is enclosed in brackets, i.e.
+ * `std::vector<int>{3, 1, 33, 7}` is printed as "[3,1,33,7]".
+ */
+template <std::ranges::InputRange rng_t>
+inline debug_stream_t & operator<<(debug_stream_t & s, rng_t && r)
+//!\cond
+    requires requires (reference_t<rng_t> l) { { debug_stream << l }; }
```

Comment:
> does the range overload not need 

no, we want to overload this also for view, because views of `char` are now printed in string-notation.

> maybe try to print a view in a test.

it works, I added an example